### PR TITLE
Convert validateFormConfirmPassword from JS to TS

### PR DESCRIPTION
**Pull request by wariofdz**
**Changed:**

- Transformed passwordMinLength data file from JS to TS because I needed it in the validateFormConfirmPassword file.
- Transformed validateFormConfirmPassword validation file from JavaScript to TypeScript file.
- Added new type Either to validations.
- Changed useRegisterConfirm hook and his test useRegisterConfirm.
- Renamed validateFormConfirmPassword to validateConfirmPassword module.

### CODE REVIEW

File: packages/frontend/web-ui/src/common/helpers/validateFormConfirmPassword.ts [hunk 1]
```diff
@@ -5,12 +5,13 @@ import { PASSWORD_MIN_LENGTH } from './passwordMinLength';
  * @param {String} password
  * @param {String} confirmPassword
  */
+
 export const validateFormConfirmPassword = (
-  formChecked,
-  password,
-  confirmPassword,
+  formChecked: { confirmPassword: string },
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/common/helpers/validateFormConfirmPassword.ts:10]: This type doesn't seem to be complete. We should reconsider this parameter when using typescript. Wouldn't it be cleaner to provide the error as a return type instead with the error messages?

**Review by notaphplover — CHANGES_REQUESTED** [review]
Nice, JSDoc must be removed though.

I honestly think the method would be way simpler if, instead of receiving a `formChecked` param, we return the validation result. The `Either` pattern would be great in this case.

File: packages/frontend/web-ui/src/common/helpers/validateFormConfirmPassword.ts (hunk 1 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/common/helpers/validateFormConfirmPassword.ts]: Done! I've applied differents changes on the code. Could you revise it?

File: packages/frontend/web-ui/src/common/helpers/validateFormConfirmPassword.ts [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
-) => {
-  let errorMessages = [];
+  password: string,
+  confirmPassword: string,
+): Either<string[], undefined> => {
+  let errorMessages: string[] = [];
 
   if (confirmPassword.length < PASSWORD_MIN_LENGTH) {
-    errorMessages.push('Password must be of minimum 6 characters length');
+    errorMessages.push('Password must be of minimum 6 characters length.');
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/common/helpers/validateFormConfirmPassword.ts:11]:
```suggestion
    errorMessages.push(`Password must be of minimum ${PASSWORD_MIN_LENGTH} characters length.`);
```

**Review by notaphplover — CHANGES_REQUESTED** [review]
Great, just a minor suggestion

File: packages/frontend/web-ui/src/common/helpers/validateFormConfirmPassword.ts (hunk 2 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/common/helpers/validateFormConfirmPassword.ts]: Done!

**Review by notaphplover — APPROVED** [review]

**Pull request merged**